src/utils/handleArguments.js

```diff
@@ -0,0 +1,254 @@
+ * @property {Array<any>} [array] - Any array.
+ * @property {HTMLMediaElement} [audio] - Both video and audio-only elements will be assigned to the audio property.
+ * @property {HTMLVideoElement} [video] - Video elements also get their own property.
+ * @property {InputImage} [image] - Any video, image, or image data.
+ */
+
+/**
+ * @class ArgHelper
+ * @implements {StandardArguments}
+ */
+class ArgHelper {
+  /**
+   * Arguments used to CREATE an image-based model can be:
+   *  - video: an HTMLVideoElement or p5 video.
+   *  - options: an object of options specific to this model.
+   *  - callback: a function to run once the model has been loaded. Called with arguments (error) or (error, result).
+   *  - modelName: some models accept a model name or URL as an argument.
+   *
+   * Arguments used to CALL a method an image-based model can be:
+   *  - image: an image or video element or an ImageData object.  Valid types: HTMLImageElement, HTMLCanvasElement,
+   *    HTMLVideoElement, ImageData, p5 image, p5 video.
+   *  - options: an object of options specific to this model.
+   *  - callback: a function to run once the method has been completed.
+   *
+   * Expected to be provided in order modelName, video/image, options, callback with any omitted.
+   * This function does not actually require any particular order.
+   *
+   * Later arguments will override earlier ones, so `this.video` should always be the first when providing arguments
+   * from a class method call.
+   *
+   *  @param {any[]} [args]
+   */
+  constructor(...args) {
+    args.forEach((arg) => this.addArg(arg));
+  }
+
+  /**
+   * Can add arguments through the constructor or at any time after construction.
+   *
+   * @param {any} arg
+   */
+  addArg(arg) {
+    // skip over falsey arguments and don't throw any error, assuming that these are omissions
+    // do this check first to prevent accessing properties on null, which is an object
+    if (arg === undefined || arg === null) {
+      return;
+    }
+    switch (typeof arg) {
+      case "string":
+        this.string = arg;
+        break;
+      case "number":
+        this.number = arg;
+        break;
+      case "function":
+        this.callback = arg
+        break;
+      case "object": {
+        if (isTensor3D(arg) || isImageData(arg)) {
+          this.image = arg;
+        }
+        // Handle p5 object and HTML elements.
+        const element = getImageElement(arg);
+        if (element) {
+          this.image = element;
+          // Videos are also both images and audio.
+          if (isVideo(element)) {
+            this.audio = element
+            this.video = element;
+          }
+        }
+          // TODO: handle audio elements and p5.sound
+        // Check for arrays
+        else if (Array.isArray(arg)) {
+          this.array = arg;
+        }
+        // All other objects are assumed to be options.
+        else {
+          this.options = arg;
```

Comment:
Question:
* If multiple objects are passed, should we spread all those obj properties into this `options` object? 😬

Reply:
That's an interesting idea!  I wouldn't want to encourage a situation where the user is providing multiple `options` objects.  But it does open up some interesting possibilities for internal use:
- We can pass the `DEFAULT_OPTIONS` as the first argument when parsing the arguments for constructing a model.  The `ArgHelper` would handle merging the provided arguments with the defaults and we could take that logic out of a bunch of constructors.
- We can merge this.options with additional options that are provided to a specific method. (For models like `BodyPix` where we are accepting options on methods like `segment`)

Reply:
With my latest change they will get a warning if there are multiple options objects.  I think that what you are suggesting is a good idea for a future PR.  But for now this one should be good to go as-is!